definitions/checks/foreman/check_checkpoint_segments.rb

```diff
@@ -0,0 +1,55 @@
+module Checks
+  module Foreman
+    class CheckpointSegments < ForemanMaintain::Check
+      metadata do
+        label :check_checkpoint_segments
+        description 'Check if checkpoint_segments parameter persent in custom hiera'
+      end
+
+      def run
+        files = []
+        files << check_postgres_config
+        files << check_custom_hiera
+        # Make sure array does not contain nil value
+        unless files.compact.empty?
+          failure_message = <<-MESSAGE.strip_heredoc
+          ERROR: Tuning option 'checkpoint_segments' found.
+          This option is no longer valid for PostgreSQL 9.5 and onwards.
+          Please remove it from following files and re-run the command.
+          #{files.join("\n")}
+          MESSAGE
+          fail! failure_message
+        end
+      end
+
+      def check_custom_hiera
+        hiera_file = feature(:installer) ? feature(:installer).custom_hiera_file : nil
```

Comment:
For what it's worth: the installer also throws an error if it finds this key. See https://github.com/theforeman/foreman-installer/commit/3306944761dd559d8343e1748a82d8021d1f99ad

Reply:
@ekohl The installer has the check but foreman maintain should take care of checkpoint_segments in the pre-upgrade check. 
Scenario - Suppose the user perform # foreman-maintain upgrade run --target-version 6.8 --assumeyes  It is expected that satellite should upgrade without any error. So it is necessary to detect and remove "checkpoint_segments" in per-upgrade.

Reply:
As a reviewer I always like to have a complete picture with references. That makes it easy to compare when I need to.

Reply:
Please see that particular commit which I linked. We also warn that there's manual tuning and encourage users to convert to `--tuning`. In the past there was a tuning guide that suggested to paste options in custom-hiera.yaml which included checkpoint_segments. We want to encourage users to use `--tuning` and drop the old options. That's why the last paragraph is there.

Reply:
Added/Copied the same error message from the commit which you mentioned.